test/factories/users.rb

```diff
@@ -0,0 +1,14 @@
+# frozen_string_literal: true
+
+FactoryBot.define do
+  factory :yocchan, class: User do
+    sequence(:email) { |n| "yocchan#{n}@example.com" }
+    password { 'password' }
+  end
+
+  factory :takeru, class: User do
+    sequence(:email) { |n| "takeru#{n}@example.com" }
+    password { 'password' }
+    name { 'たける' }
+  end
```

Comment:
```suggestion
  factory :user do
    sequence(:email) { |n| "user#{n}@example.com" }
    password { 'password' }
  end
```

事前に必要なデータを全部登録しておくFixturesとは異なり、FactoryBotを使う場合は実行するテストケースに応じて柔軟にテストデータを登録できるので、汎用的なファクトリを定義しておくことが多いですね。

特殊なデータをファクトリとして定義してしまうと、テストコードを読むときに「このファクトリにはどんなデータが入ってるんだ？」とファクトリの定義を見に行かなければならず、テストコードの可読性が低下します。

Reply:
> 汎用的なファクトリを定義しておくことが多いですね。

教えていただきありがとうございます！🙇‍♂️
違う先輩エンジニアさんにもそう言われました。奇遇ですね笑